src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -54,13 +56,19 @@ export class HttpConnection implements IConnection {
     private readonly logger: ILogger;
     private readonly options: IHttpConnectionOptions;
     private transport?: ITransport;
+    private transferFormat?: TransferFormat;
     private startPromise?: Promise<void>;
     private stopError?: Error;
     private accessTokenFactory?: () => string | Promise<string>;
 
     public readonly features: any = {};
     public onreceive: ((data: string | ArrayBuffer) => void) | null;
     public onclose: ((e?: Error) => void) | null;
+    public onreconnecting: ((e?: Error) => void) | null;
+
+    // REVIEW: I would like to have a connectionId param here to highlight it will change,
```

Comment:
I like this idea

Reply:
Aren't there are times when the client just won't have the connectionId? e.g. an automatic, non-negotiated (or was it no handshake?) websockets connection

Reply:
Yeah, connecting with WebSockets and skipping negotiation. The client never gets a negotiation response which is where they would get their connection id from the server.  

> or was it no handshake?

Handshake is always required to connect. 😄